storage/mongodb/mongo.go

```diff
@@ -0,0 +1,202 @@
+	}
+
+	log.Info("----DATABASE CONNECTED SUCCESSFULLY")
+
+	return client, nil
+}
+
+func disconnectFromDB(client *mongo.Client) {
+	// Close the connection when you're done
+	err := client.Disconnect(context.Background())
+	if err != nil {
+		return
+	}
+}
+
+func (db *MongoDB) Create(data models.DataObject) error {
+	var contact Contact
+
+	client, conncetionErr := connectToDB(db.databaseURL)
+	if conncetionErr != nil {
+		return fmt.Errorf("failed to connect to database: %w", conncetionErr)
+	}
+
+	defer disconnectFromDB(client)
+
+	// Access the database and collection
+	database := client.Database("test_contacts")
+	collection := database.Collection("contacts")
+
+	for key, value := range data {
+		contact = Contact{Phone: key, Name: value}
+	}
+
+	_, err := collection.InsertOne(context.Background(), contact)
+	if err != nil {
+		return fmt.Errorf("failed to insert item %w", err)
+	}
+
+	return nil
+}
+
+func (db *MongoDB) Read(number string) (models.DataObject, error) {
+	client, conncetionErr := connectToDB(db.databaseURL)
+	if conncetionErr != nil {
+		return nil, fmt.Errorf("failed to connect to database: %w", conncetionErr)
+	}
+
+	defer disconnectFromDB(client)
+
+	database := client.Database("test_contacts")
+	collection := database.Collection("contacts")
+
+	// Find a document
+	filter := bson.M{"phone": number}
+	var result Contact
+
+	err := collection.FindOne(context.Background(), filter).Decode(&result)
+	if err != nil {
+		return models.DataObject{}, fmt.Errorf("failed to find item %w", err)
+	}
+
+	contact := models.DataObject{
+		result.Phone: result.Name,
+	}
+
+	return contact, nil
+}
+
+func (db *MongoDB) Update(newData models.DataObject) error {
+	client, conncetionErr := connectToDB(db.databaseURL)
+	if conncetionErr != nil {
+		return fmt.Errorf("failed to connect to database: %w", conncetionErr)
+	}
+
+	defer disconnectFromDB(client)
+
+	var phoneNumber string
+	var newName string
+
+	for key, value := range newData {
+		phoneNumber = key
+		newName = value
+	}
+
+	// Access the database and collection
+	database := client.Database("test_contacts")
+	collection := database.Collection("contacts")
+
+	// Update a document
+	filter := bson.M{"phone": phoneNumber}
+	update := bson.M{"$set": bson.M{"name": newName}}
+
+	_, err := collection.UpdateOne(context.Background(), filter, update)
+	if err != nil {
+		return fmt.Errorf("failed to update item %w", err)
+	}
+
+	return nil
+}
+
+func (db *MongoDB) Delete(number string) error {
+	client, conncetionErr := connectToDB(db.databaseURL)
+	if conncetionErr != nil {
+		return fmt.Errorf("failed to connect to database: %w", conncetionErr)
+	}
+
+	defer disconnectFromDB(client)
+
+	// Access the database and collection
+	database := client.Database("test_contacts")
+	collection := database.Collection("contacts")
+
+	filter := bson.M{"phone": number}
+
+	_, err := collection.DeleteOne(context.Background(), filter)
+	if err != nil {
```

Comment:
When dealing with DBs, a common type of error for all operations is some variation of "Not Found". As such, this is a good opportunity to introduce a package-level error handler that checks for this type of error otherwise we would have to check for it every time